addons/mail/models/mail_channel.py

```diff
@@ -775,7 +775,7 @@ def _execute_channel_pin(self, pinned=False):
         """ Hook for website_livechat channel unpin and cleaning """
         self.ensure_one()
         channel_partners = self.env['mail.channel.partner'].search(
-            [('partner_id', '=', self.env.user.partner_id.id), ('channel_id', '=', self.id)])
+            [('partner_id', '=', self.env.user.partner_id.id), ('channel_id', '=', self.id), ('is_pinned', '!=', pinned)])
         self.env['bus.bus'].sendone((self._cr.dbname, 'res.partner', self.env.user.partner_id.id), self.channel_info('unsubscribe' if not pinned else False)[0])
```

Comment:
is the bus notification necessary when the pin state does not change? @odoo/rd-discuss

Reply:
If the client code was already aware that the channel was in that state then no it doesn't matter.

But if the method is executed even if the channel was already in that state it must mean something somewhere didn't know about that state (or called it for no reason), so I wouldn't consider safe to remove it.

I think the proper fix is to ensure to not call the method on channels for which it is not useful, instead of trying to change the method to ignore some values.